Code/Tools/AssetProcessor/native/ui/BuilderDataItem.cpp

```diff
@@ -0,0 +1,156 @@
+    const AZStd::string invalidJobTypeDisplayName{ "Invalid Job Type" };
+
+    BuilderDataItem::BuilderDataItem(
+        ItemType itemType, const AZStd::string& name,
+        AZ::s64 jobCount,
+        AZ::s64 totalDuration,
+        AZStd::weak_ptr<BuilderDataItem> parent)
+        : m_itemType(itemType)
+        , m_name(name)
+        , m_jobCount(jobCount)
+        , m_totalDuration(totalDuration)
+        , m_parent(parent)
+    {
+    }
+
+    int BuilderDataItem::ChildCount() const
+    {
+        return aznumeric_cast<int>(m_children.size());
+    }
+
+    const char* BuilderDataItem::GetName() const
+    {
+        return m_name.c_str();
+    }
+
+    AZ::s64 BuilderDataItem::GetJobCount() const
+    {
+        return m_jobCount;
+    }
+
+    AZ::s64 BuilderDataItem::GetTotalDuration() const
+    {
+        return m_totalDuration;
+    }
+
+    AZStd::shared_ptr<BuilderDataItem> BuilderDataItem::GetChild(int row) const
+    {
+        if (row >= m_children.size())
+        {
+            return nullptr;
+        }
+
+        return m_children[row];
+    }
+
+    AZStd::weak_ptr<BuilderDataItem> BuilderDataItem::GetParent() const
+    {
+        return m_parent;
+    }
+
+    AZStd::shared_ptr<BuilderDataItem> BuilderDataItem::UpdateOrInsertEntry(
+        JobType entryjobType, const AZStd::string& entryName, AZ::s64 entryJobCount, AZ::s64 entryTotalDuration)
+    {
+        //! only allowed to insert from builder, with a valid JobType
+        if (m_itemType != ItemType::Builder || entryjobType >= JobType::Max)
+        {
+            return nullptr;
+        }
+
+        // jobType is either CreateJobs or ProcessJob
+        const auto& jobType = m_children[aznumeric_cast<int>(entryjobType)];
+
+        AZStd::shared_ptr<BuilderDataItem> entry = nullptr;
+        if (jobType->m_childNameToIndex.contains(entryName))
+        {
+            entry = jobType->m_children[jobType->m_childNameToIndex[entryName]];
+            AZ::s64 jobCountDiff = entryJobCount - entry->m_jobCount;
+            AZ::s64 totalDurationDiff = entryTotalDuration - entry->m_totalDuration;
+            entry->m_jobCount = entryJobCount;
+            entry->m_totalDuration= entryTotalDuration;
+            jobType->UpdateMetrics(jobCountDiff, totalDurationDiff);
+        }
+        else
+        {
+            entry = jobType->m_children.emplace_back(
+                new BuilderDataItem(ItemType::Entry, entryName, entryJobCount, entryTotalDuration, jobType));
+            jobType->m_childNameToIndex[entryName] = aznumeric_cast<int>(jobType->m_children.size() - 1);
+            jobType->UpdateMetrics(entryJobCount, entryTotalDuration);
+        }
+
+        return entry;
+    }
+    bool BuilderDataItem::InitializeBuilder(AZStd::weak_ptr<BuilderDataItem> builderWeakPointer)
+    {
+        if (m_itemType != ItemType::Builder)
+        {
+            return false;
+        }
+
+        for (int i = 0; i < aznumeric_cast<int>(JobType::Max); ++i)
+        {
```

Comment:
why does this need to be an int?

Reply:
Would also argue that we should not have single character variable names 

"Required: Make sure that you use and name your variables appropriately. Use verbose names that accurately reflect what the stored value represents. Avoid using abbreviations and acronyms." - https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#variable-names

Reply:
@pollend I do not have a strong reason to use int, just what used to to count the number of items in enum. I learned that enum is stored with an integral type, but not necessarily int, and I am not sure what scoped enum does. 
Since I am using it as the index of an array (or with your recommended changes, index of `static constexpr AZStd::array`), should I use `AZStd::size_t` instead, or is there a better option that I don't know?

Reply:
Changed variable name to `jobTypeIndex`.